lib/compress.js

```diff
@@ -1805,6 +1805,7 @@ merge(Compressor.prototype, {
                         stat.alternative = make_node(AST_BlockStatement, stat, {
                             body: body
                         });
+                        statements[i] = stat;
                         statements[i] = stat.transform(compressor);
```

Comment:
I have no doubt it works, but these *seemingly* unnecessary added lines have me stumped. Why does it have an effect? Is the `statements` array accessed within the call `stat.transform(compressor)` somehow?

Reply:
As `statements` is part of `AST_Block.body`, replacing its element first before `.transform(compressor)` would enable the following look-up to work properly:
https://github.com/mishoo/UglifyJS2/blob/d959e0b86ff36156be2e7ec2cffdd57f1e30e8dd/lib/compress.js#L5592

Reply:
Holy cow. My brain hurts. ;-P

I wonder if there are other places in the code where children cannot be independently updated from their parent container(s). It sort of throws a monkey wrench into the TreeTransformer assumptions I had.

Reply:
Nested (continuation?) transforms are never easy 👻

This is why years ago I had a big clean-up of the `.transform(compressor)` calls, and eventually added in this wall of text:
https://github.com/mishoo/UglifyJS2/blob/d959e0b86ff36156be2e7ec2cffdd57f1e30e8dd/lib/compress.js#L213-L224

Reply:
Yes, I did notice that. It's sort of an organic recursive creature at this point. Just don't anger it! ;-)

The other option of performing a single transform at a time and popping out of the traversal before subsequent transforms would be more deterministic and predictable, but likely painfully slow.

Anyway, kudos to you for debugging this one.